synapseclient/client.py

```diff
@@ -1771,6 +1771,12 @@ def returnDict(destination):
             elif scheme == 'http' or scheme == 'https':
                 #TODO add support for username/password
                 response = requests.get(url, headers=self._generateSignedHeaders(url, {}), stream=True)
+
```

Comment:
if no 'content-disposition' header then take the 'tail' of the URL (after the "/"), taking care to drop the query string part of the URL (unlike what Bruce did!)